aptos-move/block-executor/src/executor.rs

```diff
@@ -479,9 +479,18 @@ where
             }
 
             if let Some(fee_statement) = last_input_output.fee_statement(txn_idx) {
-                let approx_output_size = block_gas_limit_type
-                    .block_output_limit()
-                    .and_then(|_| last_input_output.output_approx_size(txn_idx));
+                let approx_output_size = block_gas_limit_type.block_output_limit().and_then(|_| {
+                    last_input_output
+                        .output_approx_size(txn_idx)
+                        .map(|approx_output| {
+                            approx_output
+                                + if block_gas_limit_type.include_user_txn_size_in_block_output() {
+                                    block[txn_idx as usize].user_txn_bytes_len()
```

Comment:
nit: is there any way to avoid the array indexing? i.e., I assume `block[txn_idx as usize]` is valid here, but if it's not (in some weird edge case), we'll panic?

Reply:
line 107 is :

let txn = &signature_verified_block[idx_to_execute as usize];

that's assumed throughout already